GLASGOW | ITP-MAY-25 | MIRABELLE_MORAH | ONBOARDING - FORM CONTROLS

mirabellemorah opened the pull request:
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

Testable acceptance criteria for form-controls
Let's write out our testable criteria. Check each one off as you complete it.

- [ ] I have used HTML only.
- [x] I have not used any CSS or JavaScript.

### HTML

- [x] My form is semantic html.
- [x] All inputs have associated labels.
- [x] My Lighthouse Accessibility score is 100.
- [x] I require a valid name. I have defined a valid name as a text string of two characters or more.
- [x] I require a valid email.
- [x] I require one colour from a defined set of 3 colours.
- [x] I require one size from a defined set of 6 sizes.

## Changelist

I first created a branch and commited the original state. Next thing I changed was the HTML text. I added an extra <p>text</p> to explain the form to customers. Next, I used a lot of <br> to ensure there was enough spacing. Initally, the lighthouse score was 95% on accessibility specifically due to the size of the radio buttons which were about ~16px in size. Break lines <br> around the radio buttons for text choices were important to create additional space and improve the accessibility score to 100.  I further cross checked my work with ChatGPT to check if there was anything I was missing.

## Questions

None for now, thank you.

eminakturk commented:
Hello Mirabelle,

I have reviewed your code and I realized you have not deleted draft texts like : 
 <main>
      <form>
        <!-- write your html here-->
        <!-- 
          try writing out the requirements first as comments
        this will also help you fill in your PR message later-->
      </form>
      
      There's loads of them also I will share the mistakes I've seen too.
      
      There's two <head> elements and two <body> sections as you can see here : 
      
-------------------------------------------------------------------------------------------------------
       <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>My form exercise</title>
    <meta name="description" content="" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
  </head>
  <body>
    <header>
      <h1>Product Pick</h1>
    </header>
    <main>
      <form>
        <!-- write your html here-->
        <!-- 
          try writing out the requirements first as comments
        this will also help you fill in your PR message later-->
      </form>
    </main>
    <footer>
      <!-- change to your name-->
      <h2>By HOMEWORK SOLUTION</h2>
    </footer>
  </body>
</html>

<head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>My form exercise</title>
  <meta name="description" content="" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
</head>

-----------------------------------------------------------------------------------------

You should also use <fieldset> element to group your radio buttons.

If you also could fix the wording in the texts such as : <legend>T-shirt info</legend>

To this : <legend>Your T-shirt Preferences</legend>

To ensure best practises.

I have not seen any other errors. Well done.

mirabellemorah commented:
Hi @eminakturk 👋🏽 thanks a lot for your review!

**Things updated**

1. 1 comment "<(!)-- change to your name-->" deleted fro footer
2. T-shirt info changed to T-shirt preferences
3. Radio buttons wrapped in new fieldset.

**Concerns**

1. Only 1 draft content (as seen in point 1) existed in the commit version of the index file I submitted a PR for. I have removed 1 draft comment that and made a commit. The others you seem to have pointed out were not there.
2. _"There's two <head> elements and two <body> sections as you can see here"_. I do not see 2 head and 2 body elements. There's 1 head and 1 header element in my index.html file. I have also ran my entire file through chatgpt and it points to 1 head. These two concerns make me confused, if you could please point me in the right direction of these I'd be most grateful.

Thanks!

eminakturk commented:
Okay I see your main with two head and comment concerns, basically when I checked your code through your forked repository I can see there's no mistake but when I look at the changed files section in this PR I can see the same thing. I believe the problem is you have not commited your changes to Github through Vscode. 

Once you commit your changes through there and synchronize the changes, codes in this PR will change and you wont have no problem then.

mirabellemorah commented:
Thanks!

To troubleshoot this:

**Possible problem**
- Commit from VS Code to Github not reflecting

**What I've done**
- Made another commit and rather than using VS Code's UI, I used terminal with the command "git push origin form-controls"

**What's happening** 
- When I look at the changed files section in this PR, I can see all my commits including the first one and the most recent when I toggle between them. They all seem to be in sync. 


https://github.com/user-attachments/assets/9871ff7b-a5df-4287-8099-e43b37d5c887


Here's what I am seeing.

eminakturk commented:
I think you sorted the issue, I can see the correct code right now. 

One more thing, 

you can utilize chatgpt to find your errors in the code by copy and pasting your code.

I've just seen that you left "/" slashes in end of the codes.

Chatgpt would allow you to see these little things so you can fix them.

Except these everything looks good. Good luck!

mirabellemorah commented:
I've further run it through ChatGPT to see if there's anything else I can do to trouble shoot.

**My prompt to ChatGPT**

Check if changes from new commits does not show the made after pull request. 

- Pull request branch: https://github.com/mirabellemorah/Module-Onboarding/tree/form-controls
- Pull request: https://github.com/CodeYourFuture/Module-Onboarding/pull/596
- Last commit: https://github.com/CodeYourFuture/Module-Onboarding/pull/596/commits/86d863031a0e0322823b1b9c76b304f4b073189a

**Response from ChatGPT**
Your most recent commit, titled "Fix: added fieldset to group radio buttons", is indeed present in your pull request:
🔗 [View Commit](https://github.com/CodeYourFuture/Module-Onboarding/pull/596/commits/86d863031a0e0322823b1b9c76b304f4b073189a)

So, everything seems to be in order regarding the visibility of your commits in the PR.

mirabellemorah commented:
![Screenshot 2025-05-23 at 15 03 10](https://github.com/user-attachments/assets/eb9a7cc8-f96c-41f7-aa63-a1be6d594b29)

Hi again 👋🏽 sure thing I just ran that with ChatGPT. The slashes are for self closing tags. It's also what was used in the original code from CYF before I made any changes and commits. :) 

Thanks for looking through these. I guess I'm in the clear now please?

> I think you sorted the issue, I can see the correct code right now.
> 
> One more thing,
> 
> you can utilize chatgpt to find your errors in the code by copy and pasting your code.
> 
> I've just seen that you left "/" slashes in end of the codes.
> 
> Chatgpt would allow you to see these little things so you can fix them.
> 
> Except these everything looks good. Good luck!

eminakturk commented:
Yes I believe you are. 

Just wait till a volunteer approves your code and you're fine from then.

mirabellemorah commented:
> Yes I believe you are.
> 
> Just wait till a volunteer approves your code and you're fine from then.

are you not a volunteer?

eminakturk commented:
No, I am fellow learner such as you and doing this for Sprint 2 task as you will with your next sprint too 😄

eminakturk commented:
I deleted them because of your rude comment. The reason why I picked someone randomly because we meant to teach each other and appreciate it, not to be rude against.

mirabellemorah commented:
I took your reviews and feedback graciously, thanked you for them all and implemented where necessary, as seen in the comment receipt. My caution remains that while you review and learn in the process as we all are doing, you should only give reviews where you're 100% certain that it is a wrong piece of code. This is to avoid making your fellow learners more confused or looking for corrections where there may be none. 

I believe you must also learn to take feedback and corrections the same way you have given me feedback and corrections @eminakturk. As you can see, I am open to learning and making corrections. There is no rude comment I have left, except you choose to interprete a feedback as rude. That is beyond my circle of control.

> I deleted them because of your rude comment. The reason why I picked someone randomly because we meant to teach each other and appreciate it, not to be rude against.

eminakturk commented:
I do appreciate your comment right now and also sorry If I created any misunderstanding around here.
I only did what I was suppose to do which you will soon to be in my situation. 
I do code in 3 different languages which are Swift, C, Python.  Html and Javascript are on the way.
Also have 4 level 4 and above certifications including Computer Science, Cybersecurity, Embedded Systems, CompTIA A+.
I would say myself as a person who could introduce something to someone. 
As a human being I have not seen their "netlify bot" where CYF keeps your code within their systems which pushed me to see your oldest code not recent one which created this misunderstanding.
Again with sincerity I apologise for any inconvenience from my end and I'd appreciate If you wouldn't mention my name in your PR again. Thanks.

mirabellemorah commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,98 @@
+            <input type="radio" name="color" id="yellow" value="Yellow" required />
+            <label for="yellow">Yellow</label>
+          </div>
+
```
Mira- testing comments

mirabellemorah commented:
> @mirabellemorah - Your form looks very good and validation is working as expected 🙌 I left a small suggestion in your code for you to review.

Hi @jenny-alexander thanks so much for taking the time to look at my code :) Could you please point me in the direction of how to find your comment? I don't know how or where to find the comment you left for me in my code please.

**What I have done**

1. Gone to the PR
2. Scrolled to the files changed tab
3. Looked for any in-line comments or bubbles that looked [like this picture](https://images.app.goo.gl/cuPDTeT3UxsPq5Hu6) from Dev community
4. Ensured "show comments" was checked ✔️
5. Toggled through the "changes to all commits" tab to check for new comments in different commit versions
6. Looked at VS code to see if any < ! -- -- > comments were left
7. Also looked through the Git desktop app too and did a pull on VS code, asked ChatGPT and watched some youtube videos to see what a comment left in a code looks like and tried leaving myself an in-line comment too to see what it looked like (I later deleted my own comment)

**What is happening**
I am unable to locate the comment. Please help me out here if I am missing where it is. Thank you so much

jenny-alexander commented:
> > @mirabellemorah - Your form looks very good and validation is working as expected 🙌 I left a small suggestion in your code for you to review.
> 
> Hi @jenny-alexander thanks so much for taking the time to look at my code :) Could you please point me in the direction of how to find your comment? I don't know how or where to find the comment you left for me in my code please.
> 
> **What I have done**
> 
> 1. Gone to the PR
> 2. Scrolled to the files changed tab
> 3. Looked for any in-line comments or bubbles that looked [like this picture](https://images.app.goo.gl/cuPDTeT3UxsPq5Hu6) from Dev community
> 4. Ensured "show comments" was checked ✔️
> 5. Toggled through the "changes to all commits" tab to check for new comments in different commit versions
> 6. Looked at VS code to see if any < ! -- -- > comments were left
> 7. Also looked through the Git desktop app too and did a pull on VS code, asked ChatGPT and watched some youtube videos to see what a comment left in a code looks like and tried leaving myself an in-line comment too to see what it looked like (I later deleted my own comment)
> 
> **What is happening** I am unable to locate the comment. Please help me out here if I am missing where it is. Thank you so much

🥺 I am sorry @mirabellemorah - I added the comments and then forget to 'submit' them. My fault! Hope you can see them now?

mirabellemorah commented:
Ah @jenny-alexander you're right thanks so much! I have added a placeholder now.
